pkg/gc/config.go

```diff
@@ -63,18 +72,31 @@ func NewConfigFromConfigMapFunc(ctx context.Context) func(configMap *corev1.Conf
 			cm.AsDuration("stale-revision-timeout", &c.StaleRevisionTimeout),
 			cm.AsDuration("stale-revision-lastpinned-debounce", &c.StaleRevisionLastpinnedDebounce),
 
+			cm.AsInt64("stale-revision-maximum-generations", &c.StaleRevisionMaximumGenerations),
 			cm.AsInt64("stale-revision-minimum-generations", &c.StaleRevisionMinimumGenerations),
 		); err != nil {
 			return nil, fmt.Errorf("failed to parse data: %w", err)
 		}
 
-		if c.StaleRevisionMinimumGenerations < 0 {
-			return nil, fmt.Errorf("stale-revision-minimum-generations must be non-negative, was: %d", c.StaleRevisionMinimumGenerations)
-		}
+		if f, err := config.NewFeaturesConfigFromConfigMap(configMap); err == nil && f.ResponsiveRevisionGC == config.Enabled {
+			if c.StaleRevisionMinimumGenerations < -1 {
+				return nil, fmt.Errorf("stale-revision-minimum-generations must be > -2, was: %d", c.StaleRevisionMinimumGenerations)
+			}
```

Comment:
this  probably should be validated independent, since it's an existing old setting? Same with timeout validation?

Reply:
With responsive-gc enabled we want to allow for shorter values here.

Another option if we want it to be easier to be able to flip back and forth (if you want to set these to -1 for example and then disable the feature) is to make copies of the existing flags with new names for use with the new stuff. It's duplicated but more back-compat and cleanly separate. Could be an opportunity to name them better anyway?

Thoughts?